buildSrc/src/main/groovy/org/edx/builder/ConfigHelper.groovy

```diff
@@ -95,4 +95,49 @@ class ConfigHelper {
             it.write(new JsonBuilder(configJson).toPrettyString())
         }
     }
+
+    def generateGoogleServicesJson(applicationId) {
+        def config = fetchConfig()
+        def firebase = config.get("FIREBASE")
+        if (!firebase) {
+            return
+        }
+        if (!firebase.getOrDefault("ENABLED", false)) {
+            return
+        }
+
+        def googleServicesJsonPath = projectDir.path + "/app/"
+        new File(googleServicesJsonPath).mkdirs()
+
+        def projectInfo = [
+                project_number: firebase.getOrDefault("PROJECT_NUMBER", ""),
+                project_id    : firebase.getOrDefault("PROJECT_ID", ""),
+                storage_bucket: "${firebase.getOrDefault("PROJECT_ID", "")}.appspot.com"
+        ]
+        def clientInfo = [
+                mobilesdk_app_id   : firebase.getOrDefault("APPLICATION_ID", ""),
```

Comment:
currently, the edx mobile app uses the following configs.

```
FIREBASE:
    ENABLED: true
    ANALYTICS_SOURCE: ''
    CLOUD_MESSAGING_ENABLED: true
    PROJECT_NUMBER: ""
    PROJECT_ID: ""
    OAUTH_CLIENT_ID: ""
    CURRENT_KEY: ""
    MOBILE_SDK_APP_ID: ""
```
Some concerns:

- As we are creating the `google-service.json`, we should consider the key name from the `google-service.json` download from the Firebase console.

- Some of the keys are also used in the iOS app, so we can maintain the single properties set.

- During the migration this will reduce the effort of config migrations.

Reply:
> currently, the edx mobile app uses the following configs.
FIREBASE:
    ENABLED: true
    ANALYTICS_SOURCE: ''
    CLOUD_MESSAGING_ENABLED: true
    PROJECT_NUMBER: ""
    PROJECT_ID: ""
    OAUTH_CLIENT_ID: ""
    CURRENT_KEY: ""
    MOBILE_SDK_APP_ID: ""

I disagree with the current app field naming, it's confusing.

Reply:
> As we are creating the google-services.json, we should consider the key name from the google-services.json download from the Firebase console.

- The name `API_KEY` makes more sense because it is the parent entity for an object with `current_key`.
<img width="431" alt="Screenshot 2024-02-23 at 10 37 36" src="https://github.com/openedx/openedx-app-android/assets/127732735/a91bb60c-09a0-4c6d-9c73-db52dd068fe2">

- `MOBILE_SDK_APP_ID` is primarily an App ID, so it makes more sense to call it `APPLICATION_ID`.
I left a comment in the config file next to APPLICATION_ID to make it clearer and link it to MOBILE_SDK_APP_ID, hopefully this solves the issue.

Reply:
Once this PR is agreed upon, I will update the implementation on iOS and iOS config; we will maintain the single properties set.

Reply:
alright, let's move with the `API KEY` and `APPLICATION_ID`, just need to make sure we should follow the same approch for iOS.